cmd/api-response.go

```diff
@@ -482,51 +483,55 @@ func generateMultiDeleteResponse(quiet bool, deletedObjects []ObjectIdentifier,
 	return deleteResp
 }
 
-func writeResponse(w http.ResponseWriter, statusCode int, response []byte) {
+func writeResponse(w http.ResponseWriter, statusCode int, response []byte, mimeType string) {
```

Comment:
To avoid typos when providing mimeType arguments to cause errors in runtime, we can have a type alias which enumerates commonly used mimetypes in our code. 
e.g,
```go
type mimeType string
const (
  mimeNone mimeType = ""
  mimeJSON mimeType = "application/json"
  ...
)

func writeResponse(w http.ResponseWriter, statusCode int, response []byte, mType mimeType)
```